### PR TITLE
Changes in cloud init script for installing tmate

### PR DESCRIPTION
**Pull request by HenAmar**
# Commit Checklist

Thank you for creating a pull request! To help us review / merge this can you make sure that your PR adheres as much as possible to the following.

## The Basics

If you are adding new functionality, please provide evidence of the new functionality.

### CODE REVIEW

File: internal/cloudinit/cloudinit.go [hunk 1]
```diff
@@ -201,7 +201,14 @@ runcmd:
 {{ end }}
 - 'touch /root/.env'
 - '[ -f "/etc/environment" ] && cp "/etc/environment" /root/.env'
-- '/usr/bin/lite-engine server --env-file /root/.env > /var/log/lite-engine.log 2>&1 &'`
+- '/usr/bin/lite-engine server --env-file /root/.env > /var/log/lite-engine.log 2>&1 &'
+- 'mkdir /addon'
```
Comment (shubham149) [inline internal/cloudinit/cloudinit.go:205]: Addon volume is not used in vms. It is used specifically in k8 only.

Comment (shubham149) [reply internal/cloudinit/cloudinit.go:205]: Also, this needs to work for both linux amd64 and linux arm64

File: internal/cloudinit/cloudinit.go [hunk 2]
```diff
@@ -201,7 +201,14 @@ runcmd:
...
+- 'wget https://github.com/tmate-io/tmate/releases/download/2.4.0/tmate-2.4.0-static-linux-amd64.tar.xz -O /addon/tmate.xz'
```
Comment (shubham149) [inline internal/cloudinit/cloudinit.go:206]: This needs to work for both linux amd64 and linux arm64

**Review by tphoney — CHANGES_REQUESTED** [review]
There are a couple of things: 

1. you have not implemented this for the amazon linux build.
2. most importantly there is no way to disable this functionality. This installation block should be wrapped in a conditional. Ideally we should be able to disable/enable this with an environment variable. 
3. If we are enabling this via cloud init, as seen. Then we should remove the alternate way of installing it for drone. 
4. Are there binaries for mac ?

**Comment by HenAmar** [comment]
Hey @tphoney :

> 1. you have not implemented this for the amazon linux build. 
Added

> 2. most importantly there is no way to disable this functionality. This installation block should be wrapped in a conditional. Ideally we should be able to disable/enable this with an environment variable.
I took advantage of the existing "DRONE_TMATE_ENABLED" var and populate it all the way to use it. 

> 3. If we are enabling this via cloud init, as seen. Then we should remove the alternate way of installing it for drone.
I put the foundation for it and it should be easy to do it now, but I would like it to be done in a different scope and not part of this PR

> 4. Are there binaries for mac ?
No.

**Review by tphoney — APPROVED** [review]

**Review by eoinmcafee00 — APPROVED** [review]

**Pull request merged**